kore/src/Kore/Builtin/Bool.hs

```diff
@@ -231,44 +248,60 @@ unifyBothWith unify termLike1 operand1 operand2 = lift $ do
     unify' term1 term2 =
         Pattern.withoutTerm <$> unify term1 term2
 
+matchUnifyBoolOr ::
+    TermLike RewritingVariableName ->
+    TermLike RewritingVariableName ->
+    Maybe (BoolOr (TermLike RewritingVariableName))
```

Comment:
Do `BoolAnd` and `BoolOr` really need this type parameter? Isn't it always `TermLike RewritingVariableName`?

Reply:
From the looks of it, they probably don't anymore, I'm guessing this might be something left over from the `variable` to `RewritingVariableName` change?  In any case I'll go ahead and change those.